xain_fl/datasets/storage.py

```diff
@@ -55,7 +55,7 @@ def fetch_ndarray(url, fpath):
     """
     r = requests.get(url, stream=True)
 
-    logger.info("Fetching file %s", url)
+    logger.info(f"Fetching file {url}")
```

Comment:
benefit of structlog is that it allows to use variables key=value in logs:
`logger.info("Fetching file", url=str(url))`
Please adjust log messages accordingly.